lib/hanami/configuration.rb

```diff
@@ -113,11 +113,9 @@ def logger_instance
 
     setting :base_url, default: "http://0.0.0.0:2300", constructor: -> url { URI(url) }
 
-    def for_each_middleware(&blk)
-      stack = middleware.stack.dup
-      stack += sessions.middleware if sessions.enabled?
-
-      stack.each(&blk)
+    def middlewares
+      middleware.use(*sessions.middleware) if sessions.enabled?
```

Comment:
@timriley yes and it makes me uncomfortable. I initially thought about raising if called more than once. it's only used internally by the application when setting up the router. It may fire back in the future though. I'm not happy with how this session middleware is added anyway, so I'll go ahead and tweak it while I'm working on it. Thanks for pointing it out!

Reply:
@timriley done via dce9a7e852ce22a217cf43b5369cf0f2373ba328